KAS-2742: Updated newsletter model

sergiofenoll opened the pull request:
Updates the news item model and moves it from a lisp file to a JSON file, also updates all the necessary links & configs.

sergiofenoll committed:
refactor: update news-item model

sergiofenoll committed:
chore: update testdata.zip

Updates test DB after running the news item migration service.

sergiofenoll committed:
ci: use feature tagged build of Yggdrasil

sergiofenoll committed:
ci: use feature tagged build of newsletter-service

sergiofenoll committed:
fix: use camelCasing for search htmlContent property of news item

sergiofenoll committed:
Merge branch 'development' into feature/KAS-2742-newsletter-model

erikap commented on config/resources/newsletter-domain.json:
```diff
@@ -0,0 +1,152 @@
+        "agenda-item-treatments": {
+          "target": "agenda-item-treatment",
+          "predicate": "prov:wasDerivedFrom",
+          "cardinality": "many"
```
See discussion in https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1561#discussion_r1046195169. Must be a `"one"` relation.

erikap commented on config/resources/newsletter-domain.json:
```diff
@@ -0,0 +1,152 @@
+        }
+      },
+      "features": [
+        "no-pagination-defaults",
```
Can we get rid of this one by loading the themes in frontend like we load other codelists?

erikap commented on config/resources/newsletter-domain.json:
```diff
@@ -0,0 +1,152 @@
+      "features": [
+        "include-uri"
+      ],
+      "new-resource-base": "http://themis.vlaanderen.be/id/mailcampagne"
```
Trailing `/` is missing

sergiofenoll committed:
fix: make relation between news-item and agenda-item-treatment one-to-one

sergiofenoll committed:
Merge branch 'development' into feature/KAS-2742-newsletter-model

sergiofenoll committed:
fix: rename folder in testdata.zip

ValenberghsSven commented on config/resources/newsletter-domain.json:
```diff
@@ -0,0 +1,151 @@
+        "publication-date": {
+          "type": "datetime",
+          "predicate": "dct:issued"
+        },
```
We don't use this relation in our app.
`Themis-export-service` inserts it.
better to remove?

ValenberghsSven commented on config/resources/newsletter-domain.lisp:
```diff
@@ -1,50 +0,0 @@
-  :properties `((:text                  :string   ,(s-prefix "besluitvorming:inhoud"))
-                (:richtext              :string   ,(s-prefix "ext:htmlInhoud"))
-                (:subtitle              :string   ,(s-prefix "dbpedia:subtitle"))
-                (:mandatee-proposal     :string   ,(s-prefix "ext:voorstelVan"))
```
5000 hits in data for `ext:voorstelVan`. predicate not cleaned up (that I could find).
If we keep the data, we need to mention it somewhere?

erikap replied:
Ik heb een story gemaakt om dit op te volgen: https://kanselarij.atlassian.net/browse/KAS-3802

ValenberghsSven reviewed (commented):
There are 66140 newsitems currently
65522 of those records have `prov:generated`

64965 records are propagated (same for all graphs)

Do we have to figure out what those 600 records are that are not connected to an `agendaItemTreatment` before migrating?

The other 600 that have a treatment but are not propagated can't all be from active agendas but they should get migrated just the same (I think)

sergiofenoll commented:
Removing the migration service and rewriting the migration as separate migrations per predicate. That doesn't time out Virtuoso and removes a cumbersome extra service. These are the queries I used to verify that the number of news items match after running the migration.

Before migration:
```sparql
PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
PREFIX dbpedia: <http://dbpedia.org/ontology/>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
PREFIX prov: <http://www.w3.org/ns/prov#>

SELECT
(COUNT(DISTINCT ?nieuwsbericht) AS ?countNieuwsberichten)
(COUNT(DISTINCT ?subtitle) AS ?countSubtitle)
(COUNT(DISTINCT ?htmlContent) AS ?countHtmlContent)
(COUNT(DISTINCT ?plainText) AS ?countPlainText)
(COUNT(DISTINCT ?remark) AS ?countRemark)
(COUNT(DISTINCT ?modified) AS ?countModified)
(COUNT(DISTINCT ?agendaItemTreatment) AS ?countAgendaItemTreatment)
(COUNT(DISTINCT ?attachments) AS ?countAttachments)
WHERE {
  ?nieuwsbericht a besluitvorming:NieuwsbriefInfo .
  OPTIONAL { ?nieuwsbericht dbpedia:subtitle ?subtitle }
  OPTIONAL { ?nieuwsbericht ext:htmlInhoud ?htmlContent }
  OPTIONAL { ?nieuwsbericht besluitvorming:inhoud ?plainText }
  OPTIONAL { ?nieuwsbericht ext:opmerking ?remark }
  OPTIONAL { ?nieuwsbericht ext:aangepastOp ?modified }
  OPTIONAL { ?nieuwsbericht ^prov:generated ?agendaItemTreatment }
  OPTIONAL { ?nieuwsbericht ext:documentenVoorPublicatie ?attachments }
}
```
Results:
![image](https://user-images.githubusercontent.com/22411874/207421260-64835d9c-9cf3-4c3f-9a2e-fc223733790c.png)


After migration: 
```sparql
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX ex: <http://example.org/>
PREFIX besluitvorming: <http://data.vlaanderen.be/ns/besluitvorming#>
PREFIX dbpedia: <http://dbpedia.org/ontology/>
PREFIX dct: <http://purl.org/dc/terms/>
PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
PREFIX prov: <http://www.w3.org/ns/prov#>

SELECT
(COUNT(DISTINCT ?nieuwsbericht) AS ?countNieuwsberichten)
(COUNT(DISTINCT ?subtitle) AS ?countSubtitle)
(COUNT(DISTINCT ?htmlContent) AS ?countHtmlContent)
(COUNT(DISTINCT ?plainText) AS ?countPlainText)
(COUNT(DISTINCT ?remark) AS ?countRemark)
(COUNT(DISTINCT ?modified) AS ?countModified)
(COUNT(DISTINCT ?agendaItemTreatment) AS ?countAgendaItemTreatment)
(COUNT(DISTINCT ?attachments) AS ?countAttachments)
WHERE {
  ?nieuwsbericht a ext:Nieuwsbericht .
  OPTIONAL { ?nieuwsbericht dct:alternative ?subtitle }
  OPTIONAL { ?nieuwsbericht nie:htmlContent ?htmlContent }
  OPTIONAL { ?nieuwsbericht prov:value ?plainText }
  OPTIONAL { ?nieuwsbericht rdfs:comment ?remark }
  OPTIONAL { ?nieuwsbericht dct:modified ?modified }
  OPTIONAL { ?nieuwsbericht prov:wasDerivedFrom ?agendaItemTreatment }
  OPTIONAL { ?nieuwsbericht besluitvorming:heeftBijlage ?attachments }
}
```
Results:
![image](https://user-images.githubusercontent.com/22411874/207421144-8812db72-760e-49e0-b223-fdd883f9b148.png)

Note that the difference in count of subtitles is due to a quirk in the data for older news items. I checked with Johan and the conclusion was that we could ignore that.

sergiofenoll committed:
feat: add migrations for news items

sergiofenoll committed:
Revert "fix: rename folder in testdata.zip"

This reverts commit 5f189a1a96277bffbe174f0bb0b02a1e7918f75a.

sergiofenoll committed:
Revert "chore: update testdata.zip"

This reverts commit f49e1b983b661570329e130abf55a6bd8964c757.

sergiofenoll committed:
chore: update testdata.zip after migrating news items

erikap committed:
Add clarifying comment on publication-date of news-item

erikap committed:
Bumping Yggdrasil, agenda-approve and newsletter service
